### PR TITLE
validating jwt

### PR DESCRIPTION
**Pull request by huguesBouvier**
This PR enable the last of the workload API: Validating the JWT-SVID.
Like previous PR, this is mostly connecting the dots between already existing crates and light refactoring/bug fixing.
Content:
- One new crate: iot-edge-spiffe-agent/trust-bundle-manager: This is the "only" new feature. It simply poll the trust bundle based on the refresh hint it got from the first trust bundle. The polling logic is located in the main: iot-edge-spiffe-agent/agentd
- One "bug fix": I had forgotten to add a logic that filters the jwt_svid. If you have many identity, it is possible to request a jwt_svid for only one of them. The fix is in create_workload_jwts: 
image
- Most of the change come from the removing of SPIFFEID struct. I think it was not adding anything so I removed the useless layer of complexity.
- Small bug fix: Converting to string would in fact change the binary content of the key. Casting it back to binary would yield a different key
image
- Removed unused confusing TTL in entry. TTL is a config parameter in the server not selectable per entry.

Test: 
Unit test + manual test using client crate:
image

### CODE REVIEW

File: common/jwt-svid-validator/src/native_openssl.rs [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+
+        // Check token is not expired.
+        if claims.expiry < time {
+            return Err(Error::ExpiredToken {
+                current: time,
+                expiry: claims.expiry,
+            });
+        }
+
+        let _result: &String = claims
```
Comment (gordonwang0) [inline common/jwt-svid-validator/src/native_openssl.rs:74]: Do you need `let _result`?

**Review by gordonwang0 — APPROVED** [review]

File: common/jwt-svid-validator/src/native_openssl.rs [hunk 2]
```diff
@@ -0,0 +1,409 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use crate::error::Error;
+use crate::JWTSVIDValidator as JWTSVIDValidatorTrait;
+use core_objects::{get_epoch_time, JWTClaims, JWTHeader, JWTType, KeyType, TrustBundle, JWTSVID};
+use openssl::{bn::BigNum, nid, sha};
+
+#[derive(Default)]
+pub struct JWTSVIDValidator {}
```
Comment (varunpuranik) [inline common/jwt-svid-validator/src/native_openssl.rs:9]: q - why is this file called native_openssl.rs? Should we call it something like validator.rs?
Do you see any alternative implementations in the future?

Comment (huguesBouvier) [reply common/jwt-svid-validator/src/native_openssl.rs:9]: I'll chose a different name I cannot use validator: Validate?.
Clippy complains that the name of the struct is the same as the file name.

File: iot-edge-spiffe-agent/workload-api-server/src/lib.rs [hunk 3]
```diff
@@ -181,26 +217,291 @@ impl SpiffeWorkloadApi for WorkloadAPIServer {
...
+// let struct_obj: Struct = serde_json::from_str(struct_json.as_str()).unwrap();
+// This is supported by the crate: https://docs.rs/prost-wkt/0.3.0/prost_wkt/#.
+// However, this is not compatible with tonic that we use to create the server/client for Grpc.
+// So we have to use a more "manual" way to generate the Struct.
+//
+// We allow precision loss because value kind for u64 is not supported yet in prost.
+// Instead we cast is into a f64. Which is ok. 52bits mantissa of f64 is good enough for the code to live thousands of years
+// with 0 loss of precision
+#[allow(clippy::cast_precision_loss)]
+fn convert_claims_to_prost_type_strust(claims: &JWTClaims) -> Result<Struct, Error> {
```
Comment (nyanzebra) [inline iot-edge-spiffe-agent/workload-api-server/src/lib.rs:231]: minor: is strust supposed to be struct? or is strust some term that is relevant to spiffe that I should look up? :)

File: common/jwt-svid-validator/src/native_openssl.rs (hunk 2 above)
Comment (nyanzebra) [reply common/jwt-svid-validator/src/native_openssl.rs]: Feel free to add the 'allow' if you want to call it that and clippy is being annoying :) It is just a suggestion after all

**Review by nyanzebra — APPROVED** [review]

File: iot-edge-spiffe-agent/workload-api-server/src/lib.rs (hunk 3 above)
Comment (huguesBouvier) [reply iot-edge-spiffe-agent/workload-api-server/src/lib.rs]: Fixed. Thanks!

**Pull request merged**